addons/pos_self_order/controllers/orders.py

```diff
@@ -145,29 +145,42 @@ def get_tables(self, access_token):
 
         return tables
 
+    def _get_price_extra(self, product, selected_attributes):
+        price_extra = 0
+        for attribute in product.attribute_line_ids:
+            if attribute.display_name in selected_attributes:
+                attribute_value_name = selected_attributes[attribute.display_name]
+                value_ids = list(filter(
+                    lambda record: record.name == attribute_value_name,
+                    attribute.product_template_value_ids))
+                price_extra += value_ids[0].price_extra
+        return price_extra
+
     def _process_lines(self, lines, pos_config, pos_order_id):
         newLines = []
         pricelist = pos_config.pricelist_id
 
         for line in lines:
             product = pos_config.env['product.product'].browse(line.get('product_id'))
-            # todo take into account the price extra
             price_unit = pricelist._get_product_price(product, quantity=line.get('qty')) if pricelist else product.lst_price
+   
+            price_extra = self._get_price_extra(product, line.get('selected_attributes', {}))
+            price_unit += price_extra
 
             config_fiscal_pos = pos_config.default_fiscal_position_id
             selected_account_tax = config_fiscal_pos.map_tax(product.taxes_id) if config_fiscal_pos else product.taxes_id
 
             tax_results = selected_account_tax.compute_all(
                 price_unit,
                 pos_config.currency_id,
-                line.get('qty'),
+                1,
                 product,
             )
 
             newLines.append({
                 'price_unit': price_unit,
-                'price_subtotal': tax_results.get('total_excluded'),
-                'price_subtotal_incl': tax_results.get('total_included'),
+                'price_subtotal': line.get('qty') * tax_results.get('total_excluded'),
+                'price_subtotal_incl': line.get('qty') * tax_results.get('total_included'),
```

Comment:
The quantity should be handle by compute_all function, because it round the price in the correct way following setting and currency. 

In this case the rounding isn't correct.